### PR TITLE
Removing the subscription items on a subscription.

### PR DESCRIPTION
**Pull request by danlong005**
https://secure.prodtest.sj.vindicia.com/docs/rest/#modify-a-subscription-update

per the documentation this is how you remove items from the subscription.

https://legalshield.atlassian.net/browse/AD-1750

### CODE REVIEW

File: lib/cashbox/model/subscription.rb [hunk 1]
```diff
@@ -61,5 +61,20 @@ def add_subscription_items(subscription_items, bill_prorated)
           })
       self.class.cast(self, request.response)  
     end
+
+    def remove_subscription_items(subscription_items, bill_prorated)
+      request = Cashbox::Request.new(:post, route(vid),
```
Comment (dannyburrows) [inline lib/cashbox/model/subscription.rb]: How is this any different from `add_subscription_items`
```
def add_subscription_items(subscription_items, bill_prorated)
      request = Cashbox::Request.new(:post, route(vid), 
          {
              query: { 
                  effective_date: 'today',
                  bill_prorated_period: bill_prorated
              },
              body: {
                  id: id,
                  items: subscription_items
              }.to_json
          })
      self.class.cast(self, request.response)  
    end

    def remove_subscription_items(subscription_items, bill_prorated)
      request = Cashbox::Request.new(:post, route(vid),
        {
          query: {
            effective_date: 'next_billing',
            bill_prorated_period: bill_prorated
          },
          body: {
            id: id,
            items: subscription_items
          }.to_json
        })
      self.class.cast(self, request.response)
    end
```

Comment (dannyburrows) [reply lib/cashbox/model/subscription.rb]: I see the `effective_date` is different. If the logic is that close, can't this be moved into a `modify_subscription` and pass in the effective date as an enum?

Comment (danlong005) [reply lib/cashbox/model/subscription.rb]: hmm fair enough point.....

File: spec/integration/subscription_spec.rb [hunk 2]
```diff
@@ -35,10 +35,10 @@
       Cashbox::Subscription.new(
         id: 'sub_1234',
         metadata: {
-          "vin:MandateFlag": "1",
-          "vin:MandateVersion": "1.0",
-          "vin:MandateBankName": "The Bank NA",
-          "vin:MandateID": "1234123412341234"
+          'vin:MandateFlag': '1',
```
Comment (maulin) [inline spec/integration/subscription_spec.rb]: nooooo we use double quotes in ruby

File: spec/unit/model/subscription_spec.rb [hunk 3]
```diff
@@ -89,58 +88,112 @@
...
-    it 'passes the response to cast correctly' do
-      expect(subject.class).to have_received(:cast).with(subject, 'my data')
+      it 'passes parameters correctly' do
+        expect(subject.modify_subscription_items(subscprition_items, true)).to eql('modify called')
+      end
     end
 
-    it 'passes parameters correctly' do      
-      expect(subject.add_subscription_items(subscprition_items, true)).to eql('add called')
+    context 'removing subscription items' do
```
Comment (maulin) [inline spec/unit/model/subscription_spec.rb]: the tests in the add and remove contexts appear the same. Whats different?

Comment (maulin) [reply spec/unit/model/subscription_spec.rb]: the method doesn't seem to be doing anything different to distinguish between these two contexts in the test

Comment (danlong005) [reply spec/unit/model/subscription_spec.rb]: Ok.

Comment (danlong005) [reply spec/unit/model/subscription_spec.rb]: removed.

File: spec/integration/subscription_spec.rb (hunk 2 above)
Comment (danlong005) [reply spec/integration/subscription_spec.rb]: there was a mix in here of some " and some '... I can change it if you are really concerned. I just picked one to unify it.

Comment (maulin) [reply spec/integration/subscription_spec.rb]: yea we should use `"` everywhere. If there is a mix then use `'` inside the `"`

**Pull request merged**